### ISSUE TITLE
minimize_scalar issue

### ISSUE THREAD

**Issue by Harveyt47** [scipy/scipy#11593]
My issue is about the ``minimize_scalar`` in ``scipy``. Under an older version of ``scipy`` this code did to work (at least 1.3.1), now there is some error occurring.

#### Reproducing code example:

```
from scipy.special import j1
from scipy.optimize import minimize_scalar
res = minimize_scalar(j1, bounds=(1e-10, 1e-05), method='bounded')
res.x
```

#### Error message:

```
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-2-15220d9f4e21> in <module>
      2 from scipy.optimize import minimize_scalar
      3 
----> 4 res = minimize_scalar(j1, bounds=(1e-10, 1e-5), method='bounded')
      5 res.x

~/anaconda3/lib/python3.7/site-packages/scipy/optimize/_minimize.py in minimize_scalar(fun, bracket, bounds, args, method, tol, options)
    788         if isinstance(disp, bool):
    789             options['disp'] = 2 * int(disp)
--> 790         return _minimize_scalar_bounded(fun, bounds, args, **options)
    791     elif meth == 'golden':
    792         return _minimize_scalar_golden(fun, bracket, args, **options)

~/anaconda3/lib/python3.7/site-packages/scipy/optimize/optimize.py in _minimize_scalar_bounded(func, bounds, args, xatol, maxiter, disp, **unknown_options)
   1878             break
   1879 
-> 1880     if np.isnan(xf) or np.isnan(fx) or np.isnan(fu):
   1881         flag = 2
   1882 

UnboundLocalError: local variable 'fu' referenced before assignment

  ...
```

#### Scipy/Numpy/Python version information:

```
1.4.1 1.18.1 sys.version_info(major=3, minor=7, micro=4, releaselevel='final', serial=0)
```

**Comment by ev-br** [comment]
Line 1880 is pretty clear: `or np.isnan(fu)` has as mistyped `fun`.

**Comment by andyfaff** [comment]
A duplicate of #11207, which was fixed by #11208

**Comment by andyfaff** [comment]
I've marked that PR for 1.4.2, and as a backport candidate
